src/variable.rs

```diff
@@ -56,6 +67,44 @@ impl Variable {
     pub fn len(&self) -> usize {
         self.dimensions.iter().map(Dimension::len).product()
     }
+    /// Get endianness of the variable.
+    pub fn endian_value(&self) -> error::Result<Endianness> {
+		let mut e: nc_type = 0;
+		unsafe {
+			error::checked(nc_inq_var_endian(
+				self.ncid,
+				self.varid,
+				&mut e,
+			))?;
+		}
+		match e {
+			NC_ENDIAN_NATIVE => Ok(Endianness::Native),
+			NC_ENDIAN_LITTLE => Ok(Endianness::Little),
+			NC_ENDIAN_BIG => Ok(Endianness::Big),
+			_ => Err(e.into())
```

Comment:
Either `NC_EVARMETA` or create a string with an error description (there is an &str into error implementation)

Reply:
I’m afraid I don’t understand what you want.

Reply:
The value returned when querying the endianness is not an error type, and can't be converted by this `from` implmentation. You need to change this to either return `NC_EVARMETA.into()`, create a new error type for endianess, or return a string message.

Reply:
Let’s go for NC_EVARMETA then.